Sources/TripKitUI/cards/TKUIRoutingQueryInputCard.swift

```diff
@@ -82,15 +82,17 @@ public class TKUIRoutingQueryInputCard: TKUITableCard {
       startMode: startMode,
       inputs: TKUIRoutingQueryInputViewModel.UIInput(
         searchText: titleView.rx.searchText.map { ($0, forced: false) },
-        tappedDone: route,
+        tappedRoute: route,
         selected: selectedItem(in: tableView, dataSource: dataSource),
         selectedSearchMode: titleView.rx.selectedSearchMode,
         tappedSwap: titleView.swapButton.rx.tap.asSignal()
       )
     )
     
     viewModel.activeMode
-      .drive(titleView.rx.searchMode)
+      .asObservable()
+      .observe(on: MainScheduler.asyncInstance) // Avoid reentrancy
```

Comment:
Can you elaborate on this a bit more? Why does this avoid reentry?

Reply:
Re-entrancy in Rx means that observable A fires and triggers something, which then triggers A again *while it's still in the process of firing*, which is a bad idea.

Without this change, you get a reentry warning from Rx as the title view can change the search mode, this then triggers `activeMode` which then passes it back to the title view. That's the re-entrancy. Doing it on the `async instance` means the card tells the title view about the change *after* the title view is done firing (and it'll then just ignore it as the value didn't change).